src/online2bin/online2-tcp-nnet3-decode-faster_word_timestamps.cc

```diff
@@ -438,22 +446,29 @@ int main(int argc, char *argv[]) {
                 }
                 std::string str_start = std::to_string(times[i] * frame_shift * frame_subsampling + start_shift);
                 std::string str_end = std::to_string((times[i] + lengths[i]) * frame_shift * frame_subsampling + start_shift);
+                if (!assigned) {
+                    block_word_start = str_start;
+                    assigned = true;
+                }
                 message = message + "{\"word\":" + "\"" + word + "\"" + "," + "\"start\":" + str_start + "," + "\"end\":" + str_end + "},";
-                word_count += 1; // we want to remember the number of words in each lattice hypothesis
                 last_timestamp = (times[i] + lengths[i]) * frame_shift * frame_subsampling;
+                block_word_end = str_end;
+                word_count += 1; // we want to remember the number of words in each lattice hypothesis
               }
 
               // remove trailing comma
-              if (!message.empty()) {
+              if (!message.empty() && word_count > 0) {
```

Comment:
Will there be a situation where `message.empty()` is `true` and `word_count` is not zero or vice versa? Where I'm aiming with this question is whether checking only one of these variables isn't enough. Because it sounds like it is (judging from their names). So either check only one of them or make it more clear.

Reply:
Yes, you're right that this should not happen. I will keep only word_count > 0 condition.